dot/parachain/availability-store/availability_store.go

```diff
@@ -4,61 +4,160 @@
 package availabilitystore
 
 import (
+	"bytes"
 	"context"
 	"encoding/binary"
 	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/internal/database"
 	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/erasure"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
 const (
-	avaliableDataPrefix = "available"
+	availableDataPrefix = "available"
 	chunkPrefix         = "chunk"
 	metaPrefix          = "meta"
 	unfinalizedPrefix   = "unfinalized"
 	pruneByTimePrefix   = "prune_by_time"
+
+	tombstoneValue = " "
+
+	// Unavailable blocks are kept for 1 hour.
+	KEEP_UNAVAILABLE_FOR = time.Hour
+
+	// Finalized data is kept for 25 hours.
+	KEEP_FINALIZED_FOR = time.Hour * 25
+
+	// The pruning interval.
+	PRUNING_INTERVAL = time.Minute * 5
 )
 
+type BETimestamp uint64
+
+func (b BETimestamp) ToBytes() []byte {
+	res := make([]byte, 8)
+	binary.LittleEndian.PutUint64(res, uint64(b))
+	return res
+}
+
+type SubsystemClock struct{}
+
+func (sc *SubsystemClock) Now() BETimestamp {
+	return BETimestamp(time.Now().Unix())
+}
+
+// PruningConfig Struct holding pruning timing configuration.
+// The only purpose of this structure is to use different timing
+// configurations in production and in testing.
+type PruningConfig struct {
+	keepUnavailableFor time.Duration
+	keepFinalizedFor   time.Duration
+	pruningInterval    time.Duration
+}
+
+var DefaultPruningConfig = PruningConfig{
+	keepUnavailableFor: KEEP_UNAVAILABLE_FOR,
+	keepFinalizedFor:   KEEP_FINALIZED_FOR,
+	pruningInterval:    PRUNING_INTERVAL,
+}
+
 // AvailabilityStoreSubsystem is the struct that holds subsystem data for the availability store
 type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
 	availabilityStore   AvailabilityStore
-	//TODO: pruningConfig PruningConfig
-	//TODO: clock         Clock
+	pruningConfig       PruningConfig
+	clock               SubsystemClock
```

Comment:
Curious, do we really need a clock? In disputes-coordinator, it is only needed for few incoming messages and I've kept it simple by using `time.now()` directly.

Reply:
Think this will be more useful when pruning is fully implemented, for testing to more easily simulate short and long periods of time.

Reply:
Ah yes, for mocks. you're right! Thank you!